app/features/TransactionSlice.ts

```diff
@@ -438,6 +341,172 @@ export const updateTransactions = createAsyncThunk<
     }
 );
 
+interface State {
+    transactions: TransferTransaction[];
+    viewingShielded: boolean;
+    loadingTransactions: boolean;
+    hasMore: boolean;
+    synchronizing: boolean;
+}
+
+const transactionSlice = createSlice({
+    name: 'transactions',
+    initialState: {
+        transactions: [],
+        viewingShielded: false,
+        loadingTransactions: false,
+        hasMore: false,
+        synchronizing: false,
+    } as State,
+    reducers: {
+        setTransactions(state, update: PayloadAction<GetTransactionsOutput>) {
+            state.transactions = update.payload.transactions;
+            state.hasMore = update.payload.more;
+        },
+        setViewingShielded(state, viewingShielded) {
+            state.viewingShielded = viewingShielded.payload;
+        },
+        updateTransactionFields(state, update) {
+            const { hash, updatedFields } = update.payload;
+            const index = state.transactions.findIndex(
+                (transaction) => transaction.transactionHash === hash
+            );
+            if (index > -1) {
+                state.transactions[index] = {
+                    ...state.transactions[index],
+                    ...updatedFields,
+                };
+            }
+        },
+    },
+    extraReducers: (builder) => {
+        builder.addCase(loadTransactions.pending, (state, action) => {
+            const { requestId } = action.meta;
+            latestLoadingRequestId = requestId;
+
+            if (action.meta.arg.showLoading) {
+                state.loadingTransactions = true;
+            }
+        });
+
+        builder.addCase(updateTransactions.pending, (state) => {
+            state.synchronizing = true;
+        });
+
+        builder.addMatcher(
+            isAnyOf(loadTransactions.rejected, loadTransactions.fulfilled),
+            (state, action) => {
+                const { controller, append = false } = action.meta.arg;
+                const isLatest =
+                    action.meta.requestId === latestLoadingRequestId;
+
+                if (isFulfilled(action)) {
+                    state.hasMore = action.payload.more;
+
+                    if (append) {
+                        state.transactions.push(...action.payload.transactions);
+                    } else {
+                        state.transactions = action.payload.transactions;
+                    }
+                }
+
+                if (isLatest) {
+                    state.loadingTransactions = false;
+                }
+
+                controller?.finish();
+            }
+        );
+
+        builder.addMatcher(
+            isAnyOf(updateTransactions.rejected, updateTransactions.fulfilled),
+            (state, action) => {
+                state.synchronizing = false;
+                action.meta.arg.controller.finish();
```

Comment:
The controller on updateTransactions should only finish, when we have finished updating transactions (or it has been aborted).
Currently it seems to finish when we load the first batch of transactions.

Reply:
which also means it is not aborted, when you change the account / view

Reply:
hmm.. doesn't happen for me. The reject/fulfill of the action should happen when the update is recursively done, which happens when I run it.

Reply:
I'll show you tomorrow.

Reply:
This, along with the sync issue we talked about earlier today, should be fixed now.